base/gmp.jl

```diff
@@ -346,23 +346,27 @@ for (fJ, fC) in ((:-, :neg), (:~, :com))
     end
 end
 
-function <<(x::BigInt, c::Int)
-    c < 0 && throw(DomainError())
+function <<(x::BigInt, c::UInt)
```

Comment:
Making this `Culong` would force every caller to explicitly convert to `Culong`.